### PR TITLE
COMP: Take into account expected expression type on completion

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/3589.

### CODE REVIEW

**Comment by mchernyavsky** [comment]
Rebased onto current master.

File: src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt [hunk 1]
```diff
@@ -56,12 +62,28 @@ fun createLookupElement(
 
     if (forSimplePath && !element.canBeExported) {
         // It's visible and can't be exported = it's local
-        priority += LOCAL_PRIORITY
+        priority += LOCAL_PRIORITY_OFFSET
+    }
+
+
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt]: extra line

File: src/main/kotlin/org/rust/lang/core/completion/RsMacroCompletionProvider.kt [hunk 2]
```diff
@@ -55,7 +55,7 @@ object RsMacroCompletionProvider : CompletionProvider<CompletionParameters>() {
             leftSiblings.getOrNull(2)?.elementType != COLONCOLON ||
                 leftSiblings.getOrNull(3)?.elementType != IDENTIFIER)
 
-        collectCompletionVariants(result, forSimplePath = !is2segmentPath) { originalProcessor ->
+        collectCompletionVariants(result, !is2segmentPath) { originalProcessor ->
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsMacroCompletionProvider.kt]: Is it an intentional change?

**Comment by mchernyavsky** [comment]
Fixed.

**Pull request merged**